### DISCUSSION TITLE
Using external rules inside RuleTester

### DISCUSSION THREAD

**Discussion by ehellman** [eslint/eslint#15924] [Q&A] [answered]
Hi.

I am writing a plugin that checks to see if an eslint-disable-next-line comment has another comment above it. I have gotten the code working on AST Explorer but now I tried to use the yeoman generator to create an ESLint Plugin. I added a test that uses the external rule that I am testing:

```
code: `
function MyComponent(props) {
  var asd = useCallback(function() {
     // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props])
}      
      `
```

However, this results in an error when trying to run the test:
```
    message: "Definition for rule 'react-hooks/exhaustive-deps' was not found.",
```

Is there any workaround for this? Or anything I have to add to the plugin project to make it work?

**Comment by mdjermanovic** [comment, marked as answer]
You can use [`RuleTester#defineRule`](https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/lib/rule-tester/rule-tester.js#L485), here's an example in our codebase:

https://github.com/eslint/eslint/blob/10249ad1a961463b6b347be71c074951ab8e2652/tests/lib/rules/comma-dangle.js#L39-L60

**Reply by ehellman** [reply]
Thank you so much, this solved my problem! MVP for linking an example too! :)
